### PR TITLE
[CBRD-21799] replace lock-free circular queue

### PR DESCRIPTION
**Pull request by valerinistor**
http://jira.cubrid.org/browse/CBRD-21799

Replace old C lock-free circular queue implementation with new C++ one.

### CODE REVIEW

File: src/base/lockfree_circular_queue.hpp [hunk 1]
```diff
@@ -24,16 +24,15 @@
...
-#include <type_traits>
-
 #include <cassert>
 #include <climits>
 #include <cstddef>
 #include <cstdint>
+#include <thread>
+#include <type_traits>
+
+#include "base_flag.hpp"
```
Comment (ribram) [inline src/base/lockfree_circular_queue.hpp]: move ```#include "base_flag.hpp"``` before library includes.

File: src/executables/unittests_cqueue.c [hunk 2]
```diff
@@ -27,14 +27,17 @@
 #include <pthread.h>
```
Comment (ribram) [inline src/executables/unittests_cqueue.c]: Remove unittests_cqueue.c. and the unit test. New version of lock-free is already tested with new unit test system :).

File: unit_tests/lockfree/test_cqueue_functional.cpp [hunk 3]
```diff
@@ -41,262 +41,263 @@
 
 /* TODO: replace with atomic */
```
Comment (ribram) [inline unit_tests/lockfree/test_cqueue_functional.cpp]: delete the ``` /* TODO: replace with atomic */``` comments (there is one more above).

**Comment by eseokoh** [comment]
@ribram 

I don't think it will conflict. If there is something, I expect it will be easily resolved.

@alexstanro 

Do you mind accepting this prior to yours?

**Comment by alexstanro** [comment]
@eseokoh 

I agree with proposal. More, I think we should understand the new performance issue in develop branch before merging partial page. And to retest some patches for old performance issue in develop branch.

**Pull request merged**